packages/core/src/components/Modal/index.tsx

```diff
@@ -4,14 +4,23 @@ import Portal from '../Portal';
 import ModalInner, { ModalInnerProps } from './private/Inner';
 import { ESCAPE } from '../../keys';
 import { styleSheetModal } from './styles';
+import { Z_INDEX_MODAL } from '../../constants';
 
 export type ModalProps = ModalInnerProps & {
   /** Custom style sheet. */
   innerStyleSheet?: StyleSheet;
+  /** Z-index of the modal. */
+  zIndex?: number | 'auto';
```

Comment:
what's the difference between `auto` and `Z_INDEX_MODAL` here?

Reply:
This is saying `zIndex` can be either a number or a string `auto`. This is a typescript type union, not a default value.